src/backend/executor/execMain.c

```diff
@@ -1998,12 +1998,20 @@ InitPlan(QueryDesc *queryDesc, int eflags)
 		{
 			/*
 			 * A subplan will never need to do BACKWARD scan nor MARK/RESTORE.
-			 *
-			 * GPDB: We always set the REWIND flag, to delay eagerfree.
 			 */
 			sp_eflags = eflags
 				& (EXEC_FLAG_EXPLAIN_ONLY | EXEC_FLAG_WITH_NO_DATA);
-			sp_eflags |= EXEC_FLAG_REWIND;
+			
+			/*
+			 * GPDB: For ORCA, we always set the REWIND flag to delay
+			 * eagerfree. However, planner can generate InitPlans, which are
+			 * non-rewindable.
+			 */
+			if (estate->es_plannedstmt->planGen == PLANGEN_OPTIMIZER ||
```

Comment:
I think it's a weak solution in the context of https://github.com/arenadata/gpdb/pull/1124/files#diff-2658a1ae6156cefca01f7e0fe169073446b923caa722c61dbf33503854860458